### ISSUE TITLE
[ENHANCEMENT] Improve format Github action

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#803]
We could try this Github action for formatting:
```yml
name: Format suggestions

on:
  pull_request:

concurrency:
  # Skip intermediate builds: always.
  # Cancel intermediate builds: only if it is a pull request build.
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}

jobs:
  format:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: julia-actions/setup-julia@latest
        with:
          version: 1
      - run: |
          julia  -e 'using Pkg; Pkg.add("JuliaFormatter")'
          julia  -e 'using JuliaFormatter; format("."; verbose=true)'
      - uses: reviewdog/action-suggester@v1
        with:
          tool_name: JuliaFormatter
          fail_on_error: true
          filter_mode: added
```
used by [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/95612d0860f9356a426e3372aafce7feb4e933d0/.github/workflows/format.yml), which I think actually makes suggestions for formatting issues to change on the PR (whereas ours currently just says if there is a formatting issue, but doesn't suggest a change to make).

**Comment by emstoudenmire** [comment]
I was actually looking the other week to see if there was a way to do this. Would be good to have.

Could the action just run the formatter no matter what? Because in any case when the formatting is wrong, the next step is always to just run the formatter. Or I guess is the issue purely technical about needing to make a new commit after the formatter runs?

**Comment by mtfishman** [comment]
I was hoping that is what the above Github Action will do (make a commit to the PR with proposed changes to fix the formatting), but I haven't tested it.

Currently, the Github Action (https://github.com/ITensor/ITensors.jl/blob/8f8229a3ed56012a0252a30f49a421f157bf338a/.github/workflows/format_check.yml) does run no matter what (at every commit to a PR or the main branch), it just doesn't propose a change or make a new commit with formatting changes.

We also have this Github Action (https://github.com/ITensor/ITensors.jl/blob/8f8229a3ed56012a0252a30f49a421f157bf338a/.github/workflows/format_pr.yml) which makes a new PR periodically with formatting fixes. But it would nice if these behaviors were combined into a single action that directly aids when making a PR.

**Comment by emstoudenmire** [comment]
Yes, let's try these out. It would be great if every PR just had the formatter run on it and a new commit created automatically if the formatter results in a change.

**Comment by mtfishman** [comment]
I just realized that behavior could become annoying when developing.

For example, say you push a commit to a PR you are working on. Then the format action pushes it's own commit. You continue developing and happen to work on the same lines the format action edited. Now the next time you try to commit, you suddenly have a bunch of conflicts...

**Comment by mtfishman** [comment]
Maybe a better way would be to have a link on PRs on Github somewhere where you can manually make a commit that fixes any formatting issues.

**Comment by mtfishman** [comment]
Closed by #811.
